dawn/docs/parallel_model.rst

```diff
@@ -0,0 +1,251 @@
+
+Formal Definition of the Dawn Parallel Model
+============================================
+
+Contract to the user from HIR
+-----------------------------
+
+Dawn will generate code based on the operations passed from high-level
+intermediate representation. It's specification can be found `here`_.
+
+Dependencies
+~~~~~~~~~~~~
+
+Generally, stencil computation in dawn differentiates between horizontal
+and the vertical dimension. The assumption is that individual statements
+of a stencil are embarrassingly parallel in the horizontal. For that
+reason, there is no guarantee that any horizontal point's computation
+will be finished before any other in each user statement.
+
+Vertical Dependencies
+^^^^^^^^^^^^^^^^^^^^^
+
+Each vertical region of a stencil specifies a loop order of execution.
+Dawn will uphold the following rules:
+
+-  If the loop order is forward, any statement with a vertical
```

Comment:
I guess my main concern here is that there are only two loop orders in the vertical. I think we should keep "parallel" in the IIR level, too.

Reply:
IIR does have parallel. SIR does not since the parallelization happens in the back end where we traverse accesses to clarify if parallel is a viable strategy for the loop order

Reply:
and this means that the contract with the user is clearly defined but if our analysis shows that parallel execution can uphold that contract we can switch. 

This again boils down to the fact that parallelism is not expressed in the front end / SIR. The user can choose if the numerics dictate a loop order and specify that one and we will respect it if we need to (to make sure the results are the expected ones)